main.c

```diff
@@ -990,5 +982,12 @@ int main(int argc, char *argv[]) {
 		free(box);
 	}
 
+	if (state.result.width == 0 && state.result.height == 0) {
+		fprintf(stderr, "selection cancelled\n");
+		status = EXIT_FAILURE;
+	} else {
+		print_formatted_result(&state.result, &state.outputs, format);
```

Comment:
This won't do, `print_formatted_result` needs access to `state.outputs`.

Reply:
Of course you're right. Excuse my naive approach.
Would it be acceptable approach for `print_formatted_result` (maybe rename it) to return the string instead of printing it, then printing it from the main function after destroying the outputs?

Reply:
>Of course you're right. Excuse my naive approach.

No worries!

>Would it be an acceptable approach for print_formatted_result (maybe rename it) to return the string instead of printing it, then printing it from the main function after destroying the outputs?

Yeah, this sounds reasonable. The tricky part is that we need a big enough buffer. I'd suggest looking into `open_memstream` so that libc handles all of the dynamic buffer management and we can continue to use `fprintf`.